service-proxy.js

```diff
@@ -350,9 +354,10 @@ function handleLazily(conn, reqFrame) {
         }
     }
 
-    var serviceChannel = self.channel.subChannels[serviceName];
+    // use the rd (routing delegate) or the serviceName if there was no rd set
+    var serviceChannel = self.channel.subChannels[rd || serviceName];
```

Comment:
The weird thing is that we double count the rate limiter. we shouldnt increment rate limit for service if we do not route to service :(